NikBit101/dcalc_ext
# Module extractor & Autofill

## How to run it locally and test it:

1. Make sure you have `git pull` this branch, up to date.
2. Ensure that the `package.json` file in the base has the correct starting script. It should run `npm install` and `server.js` and the `config/openssl.cnf` file.
The start line should look something like this: 
`"npm install && set OPENSSL_CONF=YOUR\\PATH\\dcalc_ext\\config\\openssl.cnf && node server.js"`
3. Connect to the University VPN.
4. Start the webapp by typing `npm start` in the console of VSCode. Then when it has told you this message:
```
> dcalc_ext@1.0.0 start        
> npm install && set OPENSSL_CONF=YOUR\\PATH\\dcalc_ext\\config\\openssl.cnf && node server.js


up to date, audited 64 packages in 2s

10 packages are looking for funding  
  run `npm fund` for details

found 0 vulnerabilities
server enabled
```
Navigate to your browser and access localhost:80
5. This should open the webapp and it should be functional. You may need to refresh the page to clear the cache if you are experiencing any oddities.
6. Use some of the following UP numbers to test the module input functionality:
```
Level 5's;
- 369163290
- 369174131 (3 modules)
- 369143394 (4 modules)

Level 6's;
- 369147309
- 369138991
- 369116657 (1 module)
```
7. Make note of whether or not it functioned as expected in the Discord chat.

### Added:

- Automatic node package install (i.e. running npm start will install all necessary packages and launch the server locally on port 80)
- Autofill of 6 modules on level 5 into HTML elements (module name and agreed mark)

Removed:

- Console logs that were used for debugging purposes

This branch had been fetched from master branch.
The up number tested was this - 369163290, despite only this limit, autofill still works.
